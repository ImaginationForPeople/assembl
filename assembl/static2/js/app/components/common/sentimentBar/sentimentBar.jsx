// @flow
import * as React from 'react';

// Helpers imports
import { I18n } from 'react-redux-i18n';
import SentimentBarIcon from '../icons/sentimentBarIcon/sentimentBarIcon';

export type Props = {
  /** Value between 0 and 1 */
  value: number
};

type LevelProps = {
  error: number,
  veryBad: number,
  bad: number,
  good: number,
  excellent: number
};

type LevelMinValueProps = {
  veryBad: number,
  bad: number,
  good: number,
  excellent: number,
  maxLimit: number
};

const LEVEL: LevelProps = {
  error: -1,
  veryBad: 0,
  bad: 1,
  good: 2,
  excellent: 3
};

const LEVEL_MIN_VALUE: LevelMinValueProps = {
  veryBad: 0,
  bad: 0.35,
  good: 0.5,
  excellent: 0.65,
  maxLimit: 1
};

const SentimentBar = ({ value }: Props) => {
  let level = LEVEL.error;
  if (value >= LEVEL_MIN_VALUE.veryBad && value <= LEVEL_MIN_VALUE.maxLimit) {
    if (value < LEVEL_MIN_VALUE.bad) {
      level = LEVEL.veryBad;
    } else if (value < LEVEL_MIN_VALUE.good) {
      level = LEVEL.bad;
    } else if (value < LEVEL_MIN_VALUE.excellent) {
      level = LEVEL.good;
    } else {
      level = LEVEL.excellent;
    }
  }

  const errorKey = 'common.loader.error';
  const errorText: string = I18n.t(errorKey);

  return level === -1 ? (
    <div className="sentimentBar">
      <p className="error">{errorText}</p>
    </div>
  ) : (
    <div className="sentimentBar">
      <SentimentBarIcon level={level} />
      <p className={`level-${level}`}>{value.toFixed(2)}</p>
    </div>
  );
};

export default SentimentBar;