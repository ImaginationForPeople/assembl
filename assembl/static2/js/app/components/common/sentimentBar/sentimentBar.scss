.sentimentBar {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  .icon {
    margin-right: 15px;
  }

  p {
    font-family: $font-family3;
    font-size: 21px;
    text-align: center;
  }

  .level-0 {
    color: $sentiment-level-0;
  }

  .level-1 {
    color: $sentiment-level-1;
  }

  .level-2 {
    color: $sentiment-level-2;
  }

  .level-3 {
    color: $sentiment-level-3;
  }

  @media screen and (min-width: $screen-lg-min) {
    img {
      max-width: 100px;
    }

    p {
      font-size: 27px;
    }
  }
}
