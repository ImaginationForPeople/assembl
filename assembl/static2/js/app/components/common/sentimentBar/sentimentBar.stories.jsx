// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { number, withKnobs } from '@storybook/addon-knobs';
/* eslint-enable */

import SentimentBar from './sentimentBar';

export const defaultSentimentBarProps = {
  value: 0.5
};

const playground = {
  value: 0.5755
};

storiesOf('Semantic Analysis|SentimentBar', module)
  .addDecorator(withKnobs)
  .add(
    'default',
    withInfo()(() => (
      <div
        style={
          { width: '200px' } // eslint-disable-line comma-dangle
        }
      >
        <SentimentBar value={defaultSentimentBarProps.value} />
      </div>
    ))
  )
  .add(
    'playground',
    withInfo()(() => (
      <div
        style={
          { width: '200px' } // eslint-disable-line comma-dangle
        }
      >
        <SentimentBar value={number('value', playground.value)} />
      </div>
    ))
  );