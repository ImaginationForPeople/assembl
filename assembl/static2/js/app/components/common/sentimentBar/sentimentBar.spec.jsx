// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import initStoryshots from '@storybook/addon-storyshots';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.3';
/* eslint-enable */

import SentimentBar from './sentimentBar';
import { defaultSentimentBarProps } from './sentimentBar.stories';

initStoryshots({
  storyKindRegex: /^SentimentBar$/
});

configure({ adapter: new Adapter() });

describe('<SentimentBar /> - with shallow', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SentimentBar {...defaultSentimentBarProps} />);
  });

  it('should render an image of a sentimentBar', () => {
    expect(wrapper.find('img')).toHaveLength(1);
  });

  it('should render the value (with 2 digits)', () => {
    expect(wrapper.contains(defaultSentimentBarProps.value.toFixed(2))).toBe(true);
  });
});