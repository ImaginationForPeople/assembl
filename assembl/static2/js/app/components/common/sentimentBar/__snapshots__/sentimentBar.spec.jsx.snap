// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots SentimentBar default 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "width": "200px",
        }
      }
    >
      <div
        className="sentimentBar"
      >
        <svg
          className="icon"
          height="198"
          viewBox="0 0 108 198"
          width="108"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M2.964 1H42c-4.55 18.356-7.161 35.445-7.835 51.267l-32.02-.284C.377 32.834.65 15.84 2.964 1z"
              stroke="#5ECC6B"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <path
              d="M2 56h32.603c0 19.132.818 36.132 2.455 51H11.387C7.797 96.91 4.669 79.91 2 56z"
              fill="#93E79D"
            />
            <path
              d="M12 110h25.286c1.719 16.628 3.808 31.628 6.267 45H30.009C24.575 144.738 18.572 129.738 12 110z"
              fill="#FBB040"
            />
            <path
              d="M31.01 158h12.945a465.217 465.217 0 0 0 11.318 39.187C46.988 186.754 38.9 173.692 31.009 158z"
              fill="#F75959"
            />
            <path
              d="M99.851 99.202H82.602c-1.049 0-1.899.837-1.899 1.87 0 1.032.85 1.868 1.9 1.868h14.124c2.4 0 4.352 1.923 4.352 4.285 0 2.363-1.952 4.285-4.352 4.285H82.602c-1.049 0-1.899.836-1.899 1.868s.85 1.87 1.9 1.87h11.969a3.05 3.05 0 0 1 2.196.919c.582.592.885 1.372.854 2.198-.056 1.597-1.48 2.896-3.17 2.896H77.697c-8.216 0-14.9-6.582-14.9-14.672V92.997c0-8.09 6.684-14.672 14.9-14.672H97.12c2.398 0 4.416 1.852 4.498 4.127a4.22 4.22 0 0 1-1.216 3.133 4.35 4.35 0 0 1-3.132 1.31H82.602c-1.049 0-1.899.836-1.899 1.869 0 1.032.85 1.869 1.9 1.869H99.85c2.4 0 4.351 1.922 4.351 4.284 0 2.363-1.952 4.285-4.351 4.285zM76.994 64.635c0-2.7 2.232-4.897 4.974-4.897s4.974 2.197 4.974 4.897v9.952h-9.245c-.235 0-.468.01-.703.019v-9.97zM108 94.917c0-3.124-1.826-5.837-4.482-7.16a7.873 7.873 0 0 0 1.893-5.437c-.153-4.264-3.873-7.733-8.292-7.733h-6.381v-9.952c0-4.761-3.934-8.635-8.77-8.635-4.836 0-8.77 3.874-8.77 8.635V75.13C65.055 77.12 59 84.373 59 92.997v13.594C59 116.74 67.387 125 77.697 125h16.755c3.774 0 6.833-2.857 6.965-6.502a6.63 6.63 0 0 0-1.18-4.035c2.74-1.294 4.638-4.051 4.638-7.24a7.904 7.904 0 0 0-1.752-4.96c2.868-1.244 4.877-4.067 4.877-7.346z"
              fill="#93E79D"
            />
          </g>
        </svg>
        <p
          className="level-2"
        >
          0.50
        </p>
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Ã—
    </button>
    <div>
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            SentimentBar
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            default
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-r8d96o eyvlbql0"
          >
            <div>
              <div>
                <div
                  style={
                    Object {
                      "paddingLeft": 18,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;
                    div
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        style
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          {
                          <span
                            style={
                              Object {
                                "color": "#666",
                              }
                            }
                          >
                            {
                            <span>
                              <span
                                style={
                                  Object {
                                    "color": "#666",
                                  }
                                }
                              >
                                width
                              </span>
                            </span>
                            : 
                            <span
                              style={
                                Object {
                                  "color": "#22a",
                                  "wordBreak": "break-word",
                                }
                              }
                            >
                              '200px'
                            </span>
                            }
                          </span>
                          }
                        </span>
                      </span>
                      
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;
                    SentimentBar
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        value
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          {
                          <span
                            style={
                              Object {
                                "color": "#a11",
                              }
                            }
                          >
                            0.5
                          </span>
                          }
                        </span>
                      </span>
                       
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    /&gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 18,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;/
                    div
                    &gt;
                  </span>
                </div>
              </div>
            </div>
            <button
              className="css-1naocv3 eva467m0"
              onClick={[Function]}
            >
              <div
                className="css-lvl6aa eva467m1"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              SentimentBar
              " Component
            </h2>
            <small>
              No propTypes defined!
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots SentimentBar playground 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "width": "200px",
        }
      }
    >
      <div
        className="sentimentBar"
      >
        <svg
          className="icon"
          height="198"
          viewBox="0 0 108 198"
          width="108"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M2.964 1H42c-4.55 18.356-7.161 35.445-7.835 51.267l-32.02-.284C.377 32.834.65 15.84 2.964 1z"
              stroke="#5ECC6B"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <path
              d="M2 56h32.603c0 19.132.818 36.132 2.455 51H11.387C7.797 96.91 4.669 79.91 2 56z"
              fill="#93E79D"
            />
            <path
              d="M12 110h25.286c1.719 16.628 3.808 31.628 6.267 45H30.009C24.575 144.738 18.572 129.738 12 110z"
              fill="#FBB040"
            />
            <path
              d="M31.01 158h12.945a465.217 465.217 0 0 0 11.318 39.187C46.988 186.754 38.9 173.692 31.009 158z"
              fill="#F75959"
            />
            <path
              d="M99.851 99.202H82.602c-1.049 0-1.899.837-1.899 1.87 0 1.032.85 1.868 1.9 1.868h14.124c2.4 0 4.352 1.923 4.352 4.285 0 2.363-1.952 4.285-4.352 4.285H82.602c-1.049 0-1.899.836-1.899 1.868s.85 1.87 1.9 1.87h11.969a3.05 3.05 0 0 1 2.196.919c.582.592.885 1.372.854 2.198-.056 1.597-1.48 2.896-3.17 2.896H77.697c-8.216 0-14.9-6.582-14.9-14.672V92.997c0-8.09 6.684-14.672 14.9-14.672H97.12c2.398 0 4.416 1.852 4.498 4.127a4.22 4.22 0 0 1-1.216 3.133 4.35 4.35 0 0 1-3.132 1.31H82.602c-1.049 0-1.899.836-1.899 1.869 0 1.032.85 1.869 1.9 1.869H99.85c2.4 0 4.351 1.922 4.351 4.284 0 2.363-1.952 4.285-4.351 4.285zM76.994 64.635c0-2.7 2.232-4.897 4.974-4.897s4.974 2.197 4.974 4.897v9.952h-9.245c-.235 0-.468.01-.703.019v-9.97zM108 94.917c0-3.124-1.826-5.837-4.482-7.16a7.873 7.873 0 0 0 1.893-5.437c-.153-4.264-3.873-7.733-8.292-7.733h-6.381v-9.952c0-4.761-3.934-8.635-8.77-8.635-4.836 0-8.77 3.874-8.77 8.635V75.13C65.055 77.12 59 84.373 59 92.997v13.594C59 116.74 67.387 125 77.697 125h16.755c3.774 0 6.833-2.857 6.965-6.502a6.63 6.63 0 0 0-1.18-4.035c2.74-1.294 4.638-4.051 4.638-7.24a7.904 7.904 0 0 0-1.752-4.96c2.868-1.244 4.877-4.067 4.877-7.346z"
              fill="#93E79D"
            />
          </g>
        </svg>
        <p
          className="level-2"
        >
          0.58
        </p>
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Ã—
    </button>
    <div>
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            SentimentBar
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            playground
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-r8d96o eyvlbql0"
          >
            <div>
              <div>
                <div
                  style={
                    Object {
                      "paddingLeft": 18,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;
                    div
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        style
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          {
                          <span
                            style={
                              Object {
                                "color": "#666",
                              }
                            }
                          >
                            {
                            <span>
                              <span
                                style={
                                  Object {
                                    "color": "#666",
                                  }
                                }
                              >
                                width
                              </span>
                            </span>
                            : 
                            <span
                              style={
                                Object {
                                  "color": "#22a",
                                  "wordBreak": "break-word",
                                }
                              }
                            >
                              '200px'
                            </span>
                            }
                          </span>
                          }
                        </span>
                      </span>
                      
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;
                    SentimentBar
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        value
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          {
                          <span
                            style={
                              Object {
                                "color": "#a11",
                              }
                            }
                          >
                            0.5755
                          </span>
                          }
                        </span>
                      </span>
                       
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    /&gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 18,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#444",
                      }
                    }
                  >
                    &lt;/
                    div
                    &gt;
                  </span>
                </div>
              </div>
            </div>
            <button
              className="css-1naocv3 eva467m0"
              onClick={[Function]}
            >
              <div
                className="css-lvl6aa eva467m1"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              SentimentBar
              " Component
            </h2>
            <small>
              No propTypes defined!
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
